src/Microsoft.AspNet.Mvc.Razor/LanguageViewLocationExpander.cs

```diff
@@ -21,36 +22,32 @@ namespace RazorWebSite
     /// </example>
     public class LanguageViewLocationExpander : IViewLocationExpander
     {
-        private const string ValueKey = "language";
-        private readonly Func<ActionContext, string> _valueFactory;
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="LanguageViewLocationExpander"/>.
-        /// </summary>
-        /// <param name="valueFactory">A factory that provides tbe language to use for expansion.</param>
-        public LanguageViewLocationExpander(Func<ActionContext, string> valueFactory)
-        {
-            _valueFactory = valueFactory;
-        }
-
         /// <inheritdoc />
         public void PopulateValues(ViewLocationExpanderContext context)
         {
-            var value = _valueFactory(context.ActionContext);
-            if (!string.IsNullOrEmpty(value))
-            {
-                context.Values[ValueKey] = value;
-            }
         }
 
         /// <inheritdoc />
         public virtual IEnumerable<string> ExpandViewLocations(ViewLocationExpanderContext context,
                                                                IEnumerable<string> viewLocations)
         {
-            string value;
-            if (context.Values.TryGetValue(ValueKey, out value))
+#if DNX451
+            var cultureInfo = Thread.CurrentThread.CurrentUICulture;
+#else
+            var cultureInfo = CultureInfo.CurrentUICulture;
+#endif
+
+            var updatedViewLocations = new List<string>();
+            while (cultureInfo != cultureInfo.Parent)
+            {
+                updatedViewLocations.AddRange(ExpandViewLocationsCore(viewLocations, cultureInfo.Name));
+                cultureInfo = cultureInfo.Parent;
```

Comment:
yes, it will result in an entry for neutral culture. If I check `cultureInfo.IsNeutralCulture` as stopping condition, we won't add the neutral culture which we want to, as a fall back in case specific culture does not exist.